src/main/java/org/opendatakit/common/persistence/CommonFieldsBase.java

```diff
@@ -165,6 +165,17 @@ public final Date getLastUpdateDate() {
     return Collections.unmodifiableList(fieldList);
   }
 
+  public final boolean hasField(DataField f) {

```

Comment:
Why not use the existing getBooleanField method?

Reply:
Never mind. I think this is fine.

Reply:
With `getBooleanField` I would have to compare the result with `null` to decide if a form has a non-null value in that field... The semantics of `hasField(DataField f)` seem more convenient too.

Reply:
Anyway, everything on this PR is subject to change after #134. 

This patches only prevent some evident data corruption situations that lead to the refresh hell